src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -725,158 +201,129 @@ private class PowerShellResult
                 cancelTask.SetCanceled();
                 return;
             }
+            finally
+            {
+                _existingRequestCancellationLock.Release();
+            }
 
-            // If filesToAnalzye is empty, nothing to do so return early.
+            // If filesToAnalyze is empty, nothing to do so return early.
             if (filesToAnalyze.Length == 0)
             {
                 return;
             }
 
-            // Create a fresh cancellation token and then start the task.
-            // We create this on a different TaskScheduler so that we
-            // don't block the main message loop thread.
-            // TODO: Is there a better way to do this?
-            s_existingRequestCancellation = new CancellationTokenSource();
-            await Task.Factory.StartNew(
-                () =>
-                    DelayThenInvokeDiagnosticsAsync(
-                        750,
-                        filesToAnalyze,
-                        _configurationService.CurrentSettings.ScriptAnalysis.Enable ?? false,
-                        s_existingRequestCancellation.Token),
-                CancellationToken.None,
-                TaskCreationOptions.None,
-                TaskScheduler.Default);
-        }
-
-        private async Task DelayThenInvokeDiagnosticsAsync(
-            int delayMilliseconds,
-            ScriptFile[] filesToAnalyze,
-            bool isScriptAnalysisEnabled,
-            CancellationToken cancellationToken)
-        {
-            // First of all, wait for the desired delay period before
-            // analyzing the provided list of files
             try
             {
-                await Task.Delay(delayMilliseconds, cancellationToken);
-            }
-            catch (TaskCanceledException)
-            {
-                // If the task is cancelled, exit directly
-                foreach (var script in filesToAnalyze)
-                {
-                    PublishScriptDiagnostics(
-                        script,
-                        script.DiagnosticMarkers);
-                }
+                // Wait for the desired delay period before analyzing the provided list of files.
+                await Task.Delay(750, ct);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4416818)